### DISCUSSION TITLE
Maximum user per realm?

### DISCUSSION THREAD

**Discussion by jasondee1992** [keycloak/keycloak#15181] [Q&A]
Just wanted to ask if what is the maximum users per realm ?

**Comment by sschu** [comment]
There is no hard limit on the maximum users per realm. The number of supported users depends on hoch much storage/processing capacity you give Keycloak. There are definitely installations with millions of users in the wild.

**Reply by souravs17031999** [reply]
@sschu , is it also true for clients ? For ex. dynamic clients created using client registration flows

**Reply by sschu** [reply]
I also don't think there is a hard limit. Normally, you would expect the number of users to be the highest entity count and Keycloak is definitely optimized for that. Ideally, you create a load test scenario and see where you end up.
